src/main/resources/application.properties

```diff
@@ -20,3 +20,6 @@ quarkus.kubernetes.service-type=load-balancer
 quarkus.http.enable-compression=true
 quarkus.http.enable-decompression=true
 quarkus.qute.strict-rendering=true
+
+quarkus.index-dependency.jakarta-ws-rs-api.group-id=jakarta.ws.rs
+quarkus.index-dependency.jakarta-ws-rs-api.artifact-id=jakarta.ws.rs-api
```

Comment:
I don't really understand why this change. Can you elaborate?

Reply:
Me neiter 🤷‍♂️ When I did the upgrade:

```text
$ ./mvnw io.quarkus.platform:quarkus-maven-plugin:3.1.3.Final:update -N
```

The build (`./mvnw verify`) was failing afterwards with the error mentioned in the PR description:

```text
Caused by: io.quarkus.builder.BuildException: 
Build failure: Build failed due to errors
	[error]: Build step io.quarkus.rest.client.reactive.deployment.RestClientReactiveProcessor#addRestClientBeans threw an exception: java.lang.RuntimeException: The class jakarta.ws.rs.core.Application is not inside the Jandex index
	at org.jboss.resteasy.reactive.common.processor.JandexUtil.isImplementorOf(JandexUtil.java:397)
	at io.quarkus.rest.client.reactive.deployment.RestClientReactiveProcessor.addRestClientBeans(RestClientReactiveProcessor.java:526)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:77)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.base/java.lang.reflect.Method.invoke(Method.java:568)
	at io.quarkus.deployment.ExtensionLoader$3.execute(ExtensionLoader.java:909)
	at io.quarkus.builder.BuildContext.run(BuildContext.java:282)
	at org.jboss.threads.ContextHandler$1.runWith(ContextHandler.java:18)
	at org.jboss.threads.EnhancedQueueExecutor$Task.run(EnhancedQueueExecutor.java:2513)
	at org.jboss.threads.EnhancedQueueExecutor$ThreadBody.run(EnhancedQueueExecutor.java:1538)
	at java.base/java.lang.Thread.run(Thread.java:833)
	at org.jboss.threads.JBossThread.run(JBossThread.java:501)
```

I'm not familiar with Jandex, so I tried following [this](https://quarkus.io/guides/cdi-reference#how-to-generate-a-jandex-index) guide from the docs. To my understanding, the `jandex-maven-plugin` doesn't help in this case (tried it anyway, indeed didn't help). But adding this `quarkus.index-dependency` entry fixed the error.

Do you have an explanation for this?

Reply:
What happens if you remove the properties?

Reply:
As I said, the aforementioned error is thrown.

To illustrate this, I have created a reproducer in a new branch [`quarkus3-java17-no-props`](https://github.com/beatngu13/quarkus-petclinic/commits/quarkus3-java17-no-props) based on this PR. First, I let GitHub Actions build all branches:

* Commit: https://github.com/beatngu13/quarkus-petclinic/commit/626d8f01a060a1a9da6ed14a3c40eab76b766fec
* Successful build: https://github.com/beatngu13/quarkus-petclinic/actions/runs/5462063332/jobs/9940899374

Then, I remove the two properties:

* Commit: https://github.com/beatngu13/quarkus-petclinic/commit/41d6fd013e48acbc497ba3a844b99f6ccbfa686b
* Failed build: https://github.com/beatngu13/quarkus-petclinic/actions/runs/5462092303/jobs/9940966983#step:4:3116

Reply:
I will try and have a look tomorrow

Reply:
This is actually a bug in Quarkus itself for which I have opened a [fix](https://github.com/quarkusio/quarkus/pull/34570).

Hopefully the fix will land in Quarkus 3.2.2 at which time we can move this PR forward